### PR TITLE
[JP Plugin] Make the current Jetpack Install view configurable

### LINKED ISSUE #20018: Configure the existing Jetpack Remote Install flow
Depends on #20017 

Reuse the existing Jetpack Remote Install flow to use the new plugin installation endpoint, skip the Jetpack Connection step, and update some copy for the UI.

Installation | Error
-|-
jetpack_install_flow | error

## Tasks

- [x] #20066 
- [x] #20108
- [x] #20130 
- [x] #20131
- [ ] #20143

## Todo
- [x] Install screen
  - [x] Performs request with the new endpoint when "Continue" is tapped (in #20130)
- [x] Loading screen
  - [x] Cancel the API request when the "Cancel" button is tapped (in #20143)
- [x] Complete screen
  - [x] Remove the Cancel button (in #20131)
  - [x] Update the subtitle text (in #20130)
  - [x] Update the primary button text to "Done" (in #20130)
- [x] Error screen
  - [x] Performs request to the new endpoint when "Retry" is tapped (in #20130)
  - [x] Contact Support button tags `origin:jp-install-full-plugin-error` as metadata (in #20143)
- [x] Adjust Tracks (in #20131)

### Others

- [x] Rename `ViewData` to `ViewModel` based on [this review comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/20066#discussion_r1104889541). (in #20143)

### PR DESCRIPTION
**Pull request by dvdchr**
Refs #20018
Depends on #20063

This PR makes the `JetpackRemoteInstallStateView` configurable, so the buttons and strings can be set differently when it's reused for the full Jetpack installation process.

Previously, the view class directly reads the `JetpackRemoteInstallState` object and reads the strings from there. To make this configurable, I've abstracted this part into a delegate protocol instead, and the `JetpackRemoteInstallViewController` will be the one responsible for providing the view with the strings and controlling the visibility state of the views.

## To test

Verify that the existing Jetpack Remote Install flow is correct and not impacted by this change.

- Create a self-hosted site, and have the Jetpack plugin deactivated or uninstalled.
- Log into the site through the site address method (username & password).
- Navigate to the Notifications tab.
- Tap "Install Jetpack", and verify that:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"Install Jetpack"`
  - 🔎 The description text is correct: `"Your website credentials will not be stored and are used only for the purpose of installing Jetpack."`
  - 🔎 The primary button, "Continue", is displayed.
- To test the error screen, set your device to airplane mode and tap "Continue".
- Verify that an error screen is shown, and:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"There was a problem"`
  - 🔎 The description text is correct: `"Jetpack could not be installed at this time."`
  - 🔎 The primary button, "Retry", is shown.
  - 🔎 The Contact Support button is shown.
- Turn off your airplane mode and tap "Retry".
- This should briefly show you a loading screen. Verify that:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"Installing Jetpack"`
  - 🔎 The description text is correct: `"Installing Jetpack on your site. This can take up to a few minutes to complete."`
  - 🔎 The primary button is hidden.
  - 🔎 A loading indicator is shown.
- After the loading completes and the complete screen is shown, verify that:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"Jetpack installed"`
  - 🔎 The description text is correct: `"Now that Jetpack is installed, we just need to get you set up. This will only take a minute."`
  - 🔎 The primary button, "Set up", is shown.

## Regression Notes
1. Potential unintended areas of impact
Might impact the strings of the current Jetpack Remote Install flow.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by pachlava** [comment]
Taking a look @dvdchr, sorry for the delay!

**Comment by dvdchr** [comment]
All good! thanks for your help @pachlava 👍🏼

**Comment by pachlava** [comment]
@dvdchr 👋 

Firstly, I started with the testing instruction provided by you - thanks for making them so detailed 👍 It all worked good, I checked both with real iPhone 13 Pro and also an iPad Air 5 Simulator.

However, I then decided to go outside the scope covered by testing instructions, and have several questions. I suspect both are just not-yet-finished scenarios, but I just wanted to make sure it's the case, and they are not expected to work yet.

**Case I (video below)**

1. I had Pressable site with "usual" JP plugin installed but deactivated
2. Connected to the website via JP app
3. Installed the plugin as described in the testing instructions
4. (this is where "out of scope" starts) I tapped "Set up" at the end, waited for web view to load, and then went back to `Notifications` without performing actions in the web view.
5. Since the `Notifications` screen was not refreshed, it still contained `Install Jetpack` button (it would've disappeared if I went to `My Site` and to `Notifications` back).
6. I tapped the button and got an error at the end (`SITE_IS_JETPACK`).

This is sort of expected, since I've tapped the button that should not be there, just wanted to communicate this case.

https://user-images.githubusercontent.com/73365754/217243440-32fccf37-7eaa-42d6-a73c-ecd1e038e662.mov

Bonus part: since I firstly tried that on real iPhone, and did not see the logs, I could not understand the reason for failure. At some point I think I reached a ban (which was device-specific, not website-specific, because I could switch to iPad and continue with the same website):

photo_2023-02-07 14 22 45


**Case II**

If I have usual JP deactivated, but an individual plugin Active (it was Jetpack Social), I see this copy first, which made sense:

Screenshot 2023-02-07 at 14 06 21

But then the installation ended with `SITE_IS_JETPACK ` error. It looks like currently the API does not differentiate between full and individual plugin in this context?

Thank you! 🙂

**Comment by dvdchr** [comment]
@pachlava 👋🏼 Thank you for the extensive tests!!

> I suspect both are just not-yet-finished scenarios 
> [...] 
> I tapped the button and got an error at the end (`SITE_IS_JETPACK`).
> [...]
> If I have usual JP deactivated, but an individual plugin Active (it was Jetpack Social) [...] the installation ended with `SITE_IS_JETPACK`  error. 

As you've suspected, this is an already existing flow. We plan to reuse & modify this flow for the individual plugin case through the next PRs. As for the `SITE_IS_JETPACK` error, this is a known error that we won't handle in this project (see: pctCYC-An-p2#not-doing-out-of-scope).

Note that this project will only be accessible if you access the problem site (i.e. only individual plugins installed) **through a WP.com login**. The test steps described in this PR aim to make sure that this doesn't introduce side effects to the existing behavior—which is currently only accessible from the site address login. 

> At some point I think I reached a ban (which was device-specific, not website-specific, because I could switch to iPad and continue with the same website)

Interesting 🤔 ... Thanks for bringing this up! I'll make sure to include this in the test plan draft later on as a note/warning for the testers.

> It looks like currently the API does not differentiate between full and individual plugin in this context?

Good observation! You are correct. The current API we use right now is flawed. Later, we will use a different one that should be able to differentiate this (but only in the WP.com login scenario, though!)

I hope that makes it a little bit clear! Happy to continue here (or in Slack) if you have more questions. 🙂

**Comment by pachlava** [comment]
> Note that this project will only be accessible if you access the problem site (i.e. only individual plugins installed) through a WP.com login. The test steps described in this PR aim to make sure that this doesn't introduce side effects to the existing behavior—which is currently only accessible from the site address login.

Oh, I did not know this, thanks for explaining! I'll go on with approval from my side now, since the mentioned issues are expected.

**Review by pachlava — APPROVED** [review]
Thanks for pinging me @dvdchr!

As we discussed above, the mentioned issues are expected at this stage of the project, and the feature works as it should be working. I did the testing kind of review, but did not review the code. From my perspective, it's good to go 👍

**Review by wargcm — APPROVED** [review]
LGTM 🚀

**Comment by dvdchr** [comment]
👋🏼 @wargcm, I've refactored the implementation a bit to switch from delegation to value-type passing instead. Can I have a quick round of review from you? I've tested the changes manually myself, and they shouldn't be any different from the previous implementation. Thanks!

File: WordPress/Classes/ViewRelated/Jetpack/Install/View/JetpackRemoteInstallStateView.swift
Comment (wargcm) [inline WordPress/Classes/ViewRelated/Jetpack/Install/View/JetpackRemoteInstallStateView.swift:8]: Since this is using MVVM to configure the view, what are your thoughts about changing this name to `JetpackRemoteInstallStateViewModel`?

**Review by wargcm — APPROVED** [review]
The struct name change isn't a huge deal so I'll leave it up to you if you want to change it or not.

LGTM! :shipit:

File: WordPress/Classes/ViewRelated/Jetpack/Install/View/JetpackRemoteInstallStateView.swift
Comment (dvdchr) [reply WordPress/Classes/ViewRelated/Jetpack/Install/View/JetpackRemoteInstallStateView.swift:8]: That's fine with me; I have no strong preference for it. I was testing out a different approach and the view model name has been used for the previous delegate pattern, so I landed on the "data". 😅 

I'll include this in a clean-up PR later on.

**Pull request merged**
